entries/jQuery.htmlPrefilter.xml

```diff
@@ -0,0 +1,30 @@
+<?xml version="1.0"?>
+<entry type="method" name="jQuery.htmlPrefilter" return="String">
+  <title>jQuery.htmlPrefilter()</title>
+  <desc>Modify and filter HTML strings passed through <a href="/category/manipulation/">jQuery manipulation methods</a>.</desc>
+  <signature>
+    <added>1.12/2.2</added>
+    <argument name="html" type="String">
+      <desc>The html string on which to operate.</desc>
+    </argument>
+  </signature>
+  <longdesc>
+    <p>This method rarely needs to be called directly. Instead, use it as an entry point to modify existing <a href="/category/manipulation/">jQuery manipulation methods</a>. For instance, to remove all scripts from incoming html strings, do this:</p>
+
+    <pre><code>
+$.htmlPrefilter = function( html ) {
+  return html.replace( /&lt;script&#91;^&lt;&#93;*&lt;\/script&gt;/g, "" );
```

Comment:
This example is broken in a few ways:
1. It overwrites the default `jQuery.htmlPrefilter`, but [should instead wrap it](https://github.com/jquery/jquery/pull/2203#discussion_r28704438) like [our unit test](https://github.com/jquery/jquery/blob/fb9472c7fbf9979f48ef49aff76903ac130d0959/test/unit/manipulation.js#L2535).
2. It isn't broad enough to function as claimed. A better regex would be `/<script(?=[\x20\t\r\n\f>])[\w\W]*?<\/script([\x20\t\r\n\f][\w\W]*?|)>/gi`, but even that isn't perfect.
3. It touches on the sensitive subject of script injection. It might be better to remove something more innocuous, like [`<del>`](https://github.com/jquery/jquery/blob/fb9472c7fbf9979f48ef49aff76903ac130d0959/test/unit/manipulation.js#L2534-L2536).

Also, since we're providing sample code anyway, I was hoping to seize this opportunity to publish a fix for gh-2668 (which is the reason why it took me so long to address gh-727):
```js
var panything = "[\\w\\W]*?",

	// Whitespace
	// https://html.spec.whatwg.org/multipage/infrastructure.html#space-character
	pspace = "[\\x20\\t\\r\\n\\f]",

	// End of tag name (whitespace or greater-than)
	pnameEnd = pspace.replace( "]", ">]" ),

	// Tag name (a leading letter, then almost anything)
	// https://html.spec.whatwg.org/multipage/syntax.html#tag-open-state
	// https://html.spec.whatwg.org/multipage/syntax.html#tag-name-state
	pname = "[a-z]" + pnameEnd.replace( "[", "[^/\\0" ) + "*",

	// Void element (end tag prohibited)
	// https://html.spec.whatwg.org/multipage/syntax.html#void-elements
	pvoidName = "(?:area|base|br|col|embed|hr|img|input|keygen|link|menuitem|meta|param|" +
		"source|track|wbr)(?=" + pnameEnd + ")",

	// Attributes (double-quoted value, single-quoted value, unquoted value, or no value)
	// https://html.spec.whatwg.org/multipage/syntax.html#attributes-2
	pattrs = "(?:" + pspace + "+[^\\0-\\x20\\x7f-\\x9f=\"'/>]+(?:" + pspace + "*=" + pspace +
		"*(?:\"" + panything + "\"|'" + panything + "'|" +
		pnameEnd.replace( "[", "[^" ) + "*(?!/)" +
		")|))*" + pspace + "*",

	// Trailing content of a close tag
	pcloseTail = "(?:" + pspace + panything + "|)",

	rspecialHtml = new RegExp(
		// Non-void element that self-closes: $1–$5
		"(<)(?!" + pvoidName + ")(" + pname + ")(" + pattrs + ")(\\/)(>)|" +
			// No-innerHTML container (element, comment, or CDATA): $6
			"(<(script|style|textarea)" + pattrs + ">" + panything + "<\\/\\7" + pcloseTail + ">|" +
			"<!--" + panything + "--)",
		"gi"
	),

	// "<"; element name; attributes; ">"; "<"; "/"; element name; ">"; no-innerHTML container
	pspecialReplacement = "$1$2$3$5$1$4$2$5$6";

jQuery.htmlPrefilter = function( html ) {
	return (html + "").replace( rspecialHtml, pspecialReplacement );
};
```

Everything above `rspecialHtml = …` could be relegated to a footnote, commented out, or omitted altogether.